src/ff/ec_prime_field.rs

```diff
@@ -0,0 +1,262 @@
+use curve25519_dalek::scalar::Scalar;
+use generic_array::GenericArray;
+use hkdf::Hkdf;
+use sha2::Sha256;
+//use rand_core::RngCore;
+use typenum::U32;
+
+use crate::{
+    ff::{Field, Serializable},
+    secret_sharing::{Block, SharedValue},
+};
+
+impl Block for Scalar {
+    type Size = U32;
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub struct Fp25519(<Self as SharedValue>::Storage);
+
+impl Fp25519 {
+    pub const ONE: Self = Self(Scalar::ONE);
+
+    //must not use with ZERO
```

Comment:
I have added an assert. It will be possible to generate zero values, however when not altering the code, it will only happen with negligible probability for input values for invert. Comparing two Fp25519 is cheap since its basically a [u8;32]